### PR TITLE
Add CONTRIBUTING.md with formatting instructions

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This is (almost) the same file as in LLAMA and mallocMC.

### CODE REVIEW

**Comment by bernhardmgruber** [comment]
@psychocoderHPC mentioned somewhere else, that we should spell out the specific version of clang-format we use. I am afraid that this might cause quite an amount of maintenance every time we upgrade to a new version of clang-format, so I refrained from that for now.

**Review by sbastrakov — DISMISSED** [review]

**Comment by BenjaminW3** [comment]
I second @psychocoderHPC`s opinion that we should mention the specific version somewhere.
clang-format versions are incompatible (new options, removed options, same options format differently).
All commercial projects I know pin the version of the formatting tool in use.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+### Bash
+First install clang-format. Instructions therefore can be found on the web.
+To format your changes since branching off develop, you can run this command in bash:
+```
+git clang-format develop
+```
+To format all code in your working copy, you can run this command in bash:
+```
+find -iname *.cpp -o -iname *.hpp | xargs clang-format -i
```
Comment (psychocoderHPC) [inline CONTRIBUTING.md]: Could you please write `clang-format-11`

**Review by psychocoderHPC — DISMISSED** [review]

**Comment by psychocoderHPC** [comment]
@bernhardmgruber could you somewhere also add a phrase that clang format 11 is mandentory.

**Comment by bernhardmgruber** [comment]
Very well then, I added the clang-format version.

**Review by BenjaminW3 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+### Bash
+First install clang-format-11. Instructions therefore can be found on the web.
+To format your changes since branching off develop, you can run this command in bash:
+```
+git clang-format-11 develop
+```
+To format all code in your working copy, you can run this command in bash:
+```
+find -iname *.cpp -o -iname *.hpp | xargs clang-format-11 -i
```
Comment (psychocoderHPC) [inline CONTRIBUTING.md:19]: This command is wrong, it must be `find -iname "*.cpp" -o -iname "*.hpp" | xargs clang-format-11 -i`

Comment (psychocoderHPC) [reply CONTRIBUTING.md:19]: `*` is in the bash evaluated but this should define only a pattern for a file.

Comment (bernhardmgruber) [reply CONTRIBUTING.md:19]: Funnily enough, the command works just fine in my bash.

Comment (psychocoderHPC) [reply CONTRIBUTING.md:19]: Ohh let's check if this issue is coming from my `zsh`

Comment (psychocoderHPC) [reply CONTRIBUTING.md:19]: Mhh it is the zshell which do not like `*` only.

Comment (j-stephan) [reply CONTRIBUTING.md:19]: Yes, `zsh` needs either the quotes or a backslash. I prefer the quotes because it is portable and looks okay.

Comment (bernhardmgruber) [reply CONTRIBUTING.md:19]: Very well then, you have my permission to fix that then :)

Comment (psychocoderHPC) [reply CONTRIBUTING.md:19]: I can do it tomorrow.

**Pull request merged**